src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<R>(
+  items: R[],
+  key: keyof R,
+): GroupsMap<R> {
+  return items.reduce((acc: GroupsMap<R>, item: R) => {
+    const groupKey:string = `${item[key]}`;
```

Comment:
type will be automatically assigned to string if the right-hand-side after `=` sign is a string / string literal
```suggestion
    const groupKey = `${item[key]}`;
```